app/controllers/user_api_keys_controller.rb

```diff
@@ -147,7 +147,7 @@ def revoke
     if current_key = request.env['HTTP_USER_API_KEY']
       request_key = UserApiKey.with_key(current_key).first
       revoke_key ||= request_key
-      if request_key && request_key.id != revoke_key.id && !request_key.scopes.include?("write")
+      if request_key && request_key.id != revoke_key.id && !request_key.scopes.any? { |s| s.name == "write" }
```

Comment:
Small suggestion. I think we shouldn't interact with the scopes directly to avoid leaking implementation details.

Reply:
Yes good idea. I have moved this logic into `user_api_key.rb`, and will refactor is some more later once I add parameter filtering.